build.gradle

```diff
@@ -118,6 +118,7 @@ subprojects {
             // JDepend manages to include a different version of itself in its own transitive closure.
             force "jdepend:jdepend:${project.jdepend.toolVersion}"
         }
+        resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
```

Comment:
@chanseokoh The problem is that Travis won't let us merge it. 

We have an annoying chicken and egg problem here though: if I move this to a separate PR, that PR will fail (because it would not have been sync'd up with the common-lib) - if I sync it, it will basically be throw away since all those classes have been refactored by this PR. If you all are ok with it this time, I'll leave the common-lib syncing and cache update all in this PR. Next time I can be more careful with the timing.